rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -1392,12 +1406,42 @@ public ResponseEntity<CollectionModel<EntityModel<Project>>> getUsedByProjectDet
             }
         }
 
-        if (listOfAttachmentUsages.isEmpty()) {
-            attachmentUsageMap = null;
+        HalResource userHalResource = attachmentUsageReleases(sw360Project, sw360User, releaseList, listOfAttachmentUsages);
+        return new ResponseEntity<>(userHalResource, HttpStatus.OK);
+    }
+
+    private HalResource attachmentUsageReleases(Project sw360Project, User sw360User, List<EntityModel<Release>> releases, List<Map<String, Object>> attachmentUsageMap)
+            throws TException {
+        ObjectMapper oMapper = new ObjectMapper();
+        Map<String, ProjectReleaseRelationship> releaseIdToUsages = sw360Project.getReleaseIdToUsage();
+        Map<String, Object> projectMap = oMapper.convertValue(sw360Project, Map.class);
+        Map<String, Object> resultMap = new HashMap<>();
+        resultMap.put("linkedProjects", (Map<String, Object>) projectMap.get("linkedProjects"));
+        resultMap.put("releaseIdToUsage", (Map<String, Object>) projectMap.get("releaseIdToUsage"));
+
+        Map<String, Object> releaseIdToUsage = (Map<String, Object>) resultMap.get("releaseIdToUsage");
+        Set<String> fieldsToRemove = new HashSet<>();
+        fieldsToRemove.add("setCreatedBy");
```

Comment:
I was expecting something like this at class level.

`public static final ImmutableSet<String> SET_OF_FIELDS_TO_REMOVE = ImmutableSet.of("setCreatedBy", "setCreatedOn", "setComment", "setReleaseRelation", "setMainlineState");`

Reply:
Understood, made the changes.

Reply:
Thanks for the changes, I was expecting it at class level, but it should be fine.